app/screens/onboarding/paginator.tsx

```diff
@@ -0,0 +1,129 @@
+import Animated, {interpolate, useAnimatedStyle} from 'react-native-reanimated';
+
+import {makeStyleSheetFromTheme} from '@utils/theme';
+
+type Props = {
+    dataLength: number;
+    theme: Theme;
+    scrollX: Animated.SharedValue<number>;
+    moveToSlide: (slideIndexToMove: number) => void;
+};
+
+const DOT_SIZE = 16;
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => ({
+    dot: {
+        height: DOT_SIZE / 2,
+        borderRadius: 5,
+        backgroundColor: theme.buttonBg,
+        marginHorizontal: DOT_SIZE / 2,
+        width: DOT_SIZE / 2,
+        opacity: 0.25,
+    },
+    fixedDot: {
+        height: DOT_SIZE / 2,
+        borderRadius: 5,
+        backgroundColor: theme.buttonBg,
+        marginHorizontal: DOT_SIZE / 2,
+        width: DOT_SIZE / 2,
+        opacity: 0.25,
+        position: 'absolute',
+    },
+    outerDot: {
+        height: DOT_SIZE,
+        borderRadius: DOT_SIZE / 2,
+        backgroundColor: theme.buttonBg,
+        marginHorizontal: 4,
+        marginTop: -4,
+        position: 'absolute',
+        width: DOT_SIZE,
+        opacity: 0.15,
+    },
+    paginatorContainer: {
+        flexDirection: 'row',
+        height: 15,
+        justifyContent: 'space-between',
+        width: 120,
+    },
+}));
+
+const Paginator = ({
+    theme,
+    dataLength,
+    scrollX,
+    moveToSlide,
+}: Props) => {
+    const styles = getStyleSheet(theme);
+    return (
+        <View style={styles.paginatorContainer}>
+            {[...Array(dataLength)].map((_, index: number) => {
+                return (
+                    <Dot
+                        key={`key-${index.toString()}`}
+                        theme={theme}
+                        moveToSlide={moveToSlide}
+                        index={index}
+                        scrollX={scrollX}
+                    />
+                );
+            })}
+        </View>
+    );
+};
+
+type DotProps = {
+    index: number;
+    scrollX: Animated.SharedValue<number>;
+    theme: Theme;
+    moveToSlide: (slideIndexToMove: number) => void;
+};
+
+// this has to be extracted as a component since the useAnimatedStyle hook cant be used inside a loop
+const Dot = ({
+    index,
+    scrollX,
+    theme,
+    moveToSlide,
+}: DotProps) => {
+    const {width} = useWindowDimensions();
+    const styles = getStyleSheet(theme);
+    const inputRange = [(index - 1) * width, index * width, (index + 1) * width];
+
+    const dotOpacity = useAnimatedStyle(() => {
+        const opacity = interpolate(
+            scrollX.value,
+            inputRange,
+            [0, 1, 0],
+        );
+
+        return {opacity};
+    });
+
+    const outerDotOpacity = useAnimatedStyle(() => {
+        const opacity = interpolate(
+            scrollX.value,
+            inputRange,
+            [0, 0.15, 0],
+        );
+
+        return {opacity};
+    });
+
+    return (
+        <TouchableOpacity
+            onPress={() => moveToSlide(index)}
```

Comment:
You are creating a new function on every render. Probably not a big deal here thought.